my_sprites.py

```diff
@@ -50,6 +50,169 @@ class Sound(Enum):
     FOOTSTEP_09 = arcade.load_sound("data/audio/rpg/footstep09.ogg")
 
 
+@unique
+class EntityType(IntEnum):
+    """
+    Player types that map to numbers in filename suffixes
+    """
+    WIZARD = 7 * 12 + 0
+    MAN_01 = 7 * 12 + 1
+    BLACKSMITH = 7 * 12 + 2
+    VIKING = 7 * 12 + 3
+    MAN_02 = 7 * 12 + 4
+    KNIGHT_CLOSED_HELMET = 8 * 12 + 0
+    KNIGHT_OPEN_HELMET = 8 * 12 + 1
+    KNIGHT_NO_HELMET = 8 * 12 + 2
+    WOMAN_YOUNGER = 8 * 12 + 3
+    WOMAN_OLDER = 8 * 12 + 4
+
+
+class Entity(arcade.Sprite):
+    """
+    parent class for both enemies and players. Features include attacks, hp and more.
+    """
+
+    def __init__(self,
+                 position: tuple[float, float],
+                 graphics_type: EntityType,
+                 max_hp: int,
+                 speed: int,
+                 window: arcade.Window,
+                 equipped_weapon=None,
+                 scale=1.0,
+                 ):
+
+        # graphics
+        # Load the image twice, with one flipped, so we have left/right facing textures
+        self.textures = arcade.load_texture_pair(f"images/tiny_dungeon/Tiles/tile_{graphics_type:0=4}.png")
+        self.texture = self.textures[0]
+
+        # hp
+        self._max_hp = max_hp
+        self._cur_hp = max_hp
+        self.healthbar = HealthBar(max_health=max_hp)
+
+        self.speed = speed
+        self.window = window
+        self._equipped_weapon = equipped_weapon
+        # angle used for the dir the sprite is facing. for the enemy this will usually be the angle to the target
+        self.facing_dir = 0
+
+        self._emotes = arcade.SpriteList()
+
+        super().__init__(scale=scale,
+                         center_x=position[0],
+                         center_y=position[1])
+
+        self.texture = self.textures[0]
+
+        # amount of seconds before the sprite can update
+        self.pause_timer = 0
+
+    @property
+    def max_hp(self):
+        return self._max_hp
+
+    @property
+    def cur_hp(self):
+        return self.cur_hp
+
+    @cur_hp.setter
+    def cur_hp(self, new_hp):
+        self.cur_hp = max(0, min(new_hp, self.max_hp))
+
+    @property
+    def emotes(self):
```

Comment:
I think this was for drawing/updating. If we do that in the class, then we should probably remove this property?